File: packages/aideo-serv/src/aideo_serv/models/task.py
"""Task data models for aideo-serv."""

from datetime import datetime, timezone
from enum import StrEnum
from uuid import UUID

from pydantic import BaseModel, Field


class TaskStatus(StrEnum):
    """Video generation task lifecycle states."""

    QUEUED = "queued"
    RUNNING = "running"
    GENERATING = "generating"
    COMPLETED = "completed"
    FAILED = "failed"
    CANCELLED = "cancelled"


class TaskCreate(BaseModel):
    """Payload for creating a new generation task."""

    prompt: str = Field(min_length=1, max_length=4096)
    params: dict | None = None
    task_type: str = "video_generation"  # video_generation | speech_to_text
    input_files: list[dict] | None = None  # e.g. [{"path": "...", "type": "audio"}]


class Task(BaseModel):
    """A generation task with full lifecycle state.

    Supports multiple task types (video_generation, speech_to_text, …)
    routed to the appropriate inference service via WebSocket.
    """

    id: UUID
    prompt: str
    params: dict | None = None
    task_type: str = "video_generation"
    input_files: list[dict] | None = None
    status: TaskStatus = TaskStatus.QUEUED
    progress: float = Field(default=0.0, ge=0.0, le=100.0)
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    updated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    result_path: str | None = None
    result_url: str | None = None
    result_data: dict | None = None
    previews: list[str] = Field(default_factory=list)
    error_message: str | None = None
    project_id: UUID | None = None


class TaskListResponse(BaseModel):
    """Paginated list of tasks."""

    tasks: list[Task]
    total: int
    offset: int
    limit: int

File: packages/aideo-ipad/Aideo/Aideo/Models/Task.swift
import Foundation

/// 任务状态枚举，与后端 TaskStatus 对齐
enum TaskStatus: String, Codable, CaseIterable {
    case queued
    case running
    case generating
    case completed
    case failed
    case cancelled

    /// 显示名称
    var displayName: String {
        switch self {
        case .queued:     "排队中"
        case .running:    "运行中"
        case .generating: "生成中"
        case .completed:  "已完成"
        case .failed:     "失败"
        case .cancelled:  "已取消"
        }
    }

    /// SF Symbol 图标
    var iconName: String {
        switch self {
        case .queued:     "hourglass"
        case .running:    "play.circle"
        case .generating: "wand.and.stars"
        case .completed:  "checkmark.circle.fill"
        case .failed:     "xmark.circle.fill"
        case .cancelled:  "slash.circle"
        }
    }

    /// 是否终态
    var isTerminal: Bool {
        self == .completed || self == .failed || self == .cancelled
    }
}

/// 任务模型，对应后端 GenerationTask
struct TaskModel: Identifiable, Codable, Equatable, Sendable {
    let id: UUID
    var prompt: String
    var params: GenerationParams?
    var status: TaskStatus
    var progress: Double
    var createdAt: Date
    var updatedAt: Date
    var resultPath: String?
    var resultURL: URL?
    var previews: [String]
    var errorMessage: String?

    // v2: 结构化提交 & 项目关联
    var projectId: UUID?
    var outputNodeId: UUID?
    var promptStructured: [String: AnyCodable]?

    enum CodingKeys: String, CodingKey {
        case id, prompt, params, status, progress
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case resultPath = "result_path"
        case resultURL = "result_url"
        case previews
        case errorMessage = "error_message"
        case projectId = "project_id"
        case outputNodeId = "output_node_id"
        case promptStructured = "prompt_structured"
    }

    static func == (lhs: TaskModel, rhs: TaskModel) -> Bool {
        lhs.id == rhs.id
    }
}

/// 任务列表响应
struct TaskListResponse: Codable, Sendable {
    let tasks: [TaskModel]
    let total: Int
    let offset: Int
    let limit: Int
}
